src/cal/mod.rs

```diff
@@ -0,0 +1,176 @@
+use std::collections::BTreeSet;
+use std::ops::Range;
+
+use chrono::DateTime;
+use chrono::Duration;
+use chrono::Utc;
+
+use std::cmp::Ordering;
+
+pub struct Cal {
+    events: BTreeSet<CmpEvent>,
+}
+
+#[allow(dead_code)]
+impl Cal {
+    pub fn new() -> Cal {
+        Cal {
+            events: BTreeSet::new(),
+        }
+    }
+
+    pub fn events_in(&self, range: Range<DateTime<Utc>>) -> impl Iterator<Item = Event> + '_ {
+        let event_range = Range {
+            start: CmpEvent::from_date(range.start),
+            end: CmpEvent::from_date(range.end),
+        };
+        self.events.range(event_range).map(|x| x.event.clone())
+    }
+
+    pub fn add_event(&mut self, event: Event) -> bool {
+        self.events.insert(CmpEvent::from_event(event))
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Event {
+    pub organizer: String,
+    pub description: String,
+    pub date: DateTime<Utc>,
+    pub duration: Duration,
+}
+
+impl Default for Event {
```

Comment:
Making a `Default` instance signals to`Event` users that there is a sensible "default" event, when there really isn't.

It looks like you only use this to create "dummy" events for event lookup and for tests. It would be better to just have a private function that creates a dummy event with these members.

You can still use the [update syntax](https://doc.rust-lang.org/1.9.0/book/structs.html#update-syntax) with such a function.